### PR TITLE
Feature/id 517/kafka to mqtt transformer

### PR DESCRIPTION
**Pull request by DC2-DanielKrueger**

### CODE REVIEW

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/TransformerInitInput.java [hunk 1]
```diff
@@ -28,4 +30,14 @@
...
 @Immutable
 @DoNotImplement
-public interface TransformerInitInput {}
+public interface TransformerInitInput {
+
+    /**
+     * Getter Method for the {@link MetricRegistry} of this HiveMQ node. It is possible to add own metrics to monitor
+     * custom business logic.
+     *
+     * @return the {@link MetricRegistry} of this HiveMQ node
```
Comment (sauroter) [inline src/main/java/com/hivemq/extensions/kafka/api/transformers/TransformerInitInput.java:39]:
```suggestion
     * @return the {@link MetricRegistry} of the HiveMQ node this Enterprise Extension for Kafka is running on
```

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttOutput.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * {@link Publish}s given to this output will be published by HiveMQ.
+ *
+ * @author Christoph Schäbel
+ * @since 4.5.0
+ */
+@DoNotImplement
+public interface KafkaToMqttOutput {
+
+    /**
+     * @return an new {@link PublishBuilder}.
```
Comment (sauroter) [inline src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttOutput.java:39]:
```suggestion
     * @return a new {@link PublishBuilder}.
```

**Review by sauroter — APPROVED** [review]

**Review by LBrandl — APPROVED** [review]

**Comment by sauroter** [comment]
The comments here need to be addressed.

File: README.adoc [hunk 3]
```diff
@@ -31,3 +31,11 @@ To use the SDK, add the following dependency to your Maven project:
 
 The HiveMQ Kafka Extension SDK is licensed under the `APACHE LICENSE, VERSION 2.0`.
 A copy of the license can be found link:LICENSE[here].
+
+== Setup
+Make sure the `hivemq extension sdk` repository is checked out in the same folder as the customization sdk, so the hivemq
```
Comment (SgtSilvio) [inline README.adoc:36]:
```suggestion
Make sure the `hivemq-extension-sdk` repository is checked out in the same folder as the customization sdk, so the hivemq
```

File: README.adoc [hunk 4]
```diff
@@ -31,3 +31,11 @@ To use the SDK, add the following dependency to your Maven project:
...
+extension sdk can be linked with this repository (see `settings.gradle.kts` file line `includeBuild("../hivemq-extension-sdk")`).
```
Comment (SgtSilvio) [inline README.adoc:37]:
```suggestion
To be able to use the latest change of the `hivemq-extension-sdk` please check out the https://github.com/hivemq/hivemq-extension-sdk repository in the same folder as the current project.
(See line `includeBuild("../hivemq-extension-sdk")` in `settings.gradle.kts`).
```

File: README.adoc [hunk 5]
```diff
@@ -31,3 +31,11 @@ To use the SDK, add the following dependency to your Maven project:
...
+
+NOTE: You can also change the path in the `settings.gradle.kts` to the path of your `hivemq extension sdk`.
```
Comment (SgtSilvio) [inline README.adoc:39]:
```suggestion
NOTE: You can also change the path in the `settings.gradle.kts` to the path of your `hivemq-extension-sdk`.
```

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttInput.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import com.hivemq.extensions.kafka.api.model.KafkaRecord;
+
+/**
+ * The input parameter of the {@link KafkaToMqttTransformer}. It contains the information of the to be transformed
+ * {@link KafkaRecord}.
+ * <p>
+ * The MqttToKafkaInput allows access to the {@link KafkaCluster}.
+ *
+ * @author Christoph Schäbel
+ * @author Geord Held
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttInput.java:32]:
```suggestion
 * @author Georg Held
```

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @param input  the {@link KafkaToMqttInput} contains the triggering {@link com.hivemq.extensions.kafka.api.model.KafkaRecord
+     *               KafkaRecord} and the {@link com.hivemq.extensions.kafka.api.model.KafkaCluster KafkaCluster}
+     *               information.
+     * @param output the {@link KafkaToMqttOutput} allows to {@link KafkaToMqttOutput#setPublishes(java.util.List)
+     *               provide a list of Publishes}. If no output is set, a empty List is used as default and the kafka
+     *               record will not be processed again, but ignored.
+     * @since 4.5.0
+     */
+    @ThreadSafe
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java:58]: Does the MqttToKafkaTransformer also need to be thread safe?

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extension.sdk.api.annotations.ThreadSafe;
+import com.hivemq.extensions.kafka.api.transformers.Transformer;
+
+/**
+ * Implement this transformer for the programmatic creation of {@link com.hivemq.extension.sdk.api.services.publish.Publish
+ * Publishes} from {@link com.hivemq.extensions.kafka.api.model.KafkaRecord KafkaRecords}. One instance of the
+ * implementing class is created per reference in the kafka-configuration.xml. The methods of this interface may be
+ * called concurrently and must be {@link com.hivemq.extension.sdk.api.annotations.ThreadSafe}.
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java:27]:
```suggestion
 * called concurrently and must be thread-safe.
```

File: src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java (hunk 7 above)
Comment (DC2-DanielKrueger) [reply src/main/java/com/hivemq/extensions/kafka/api/transformers/kafkatomqtt/KafkaToMqttTransformer.java:58]: i dont think so. i think it is only called single threaded

**Pull request merged**